pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,750 @@
-                if self.precision == 16
+                f"Using 16bit {self._amp_type_flag.value} Automatic Mixed Precision (AMP)"
+                if self._precision_flag == 16
                 else "Using bfloat16 Automatic Mixed Precision (AMP)"
             )
 
-            if self.amp_type == AMPType.NATIVE:
-                device = "cpu" if self.use_cpu else "cuda"
+            if self._amp_type_flag == AMPType.NATIVE:
+                device = "cpu" if self._accelerator_flag == "cpu" else "cuda"
 
-                if self._is_sharded_training_type:
-                    return ShardedNativeMixedPrecisionPlugin(self.precision, device)
-                if self._is_fully_sharded_training_type:
-                    return FullyShardedNativeMixedPrecisionPlugin(self.precision, device)
-                return NativeMixedPrecisionPlugin(self.precision, device)
+                if isinstance(self.strategy, (DDPShardedStrategy, DDPSpawnShardedStrategy)):
+                    return ShardedNativeMixedPrecisionPlugin(self._precision_flag, device)
+                if isinstance(self.strategy, DDPFullyShardedStrategy):
+                    return FullyShardedNativeMixedPrecisionPlugin(self._precision_flag, device)
+                return NativeMixedPrecisionPlugin(self._precision_flag, device)
 
-            if self.amp_type == AMPType.APEX:
-                if self._is_sharded_training_type or self._is_fully_sharded_training_type:
-                    raise MisconfigurationException(
-                        "Sharded plugins are not supported with apex, please switch to `amp_backend='native'`."
-                    )
-                self.amp_level = self.amp_level or "O2"
-                return ApexMixedPrecisionPlugin(self.amp_level)
+            if self._amp_type_flag == AMPType.APEX:
+                self._amp_level_flag = self._amp_level_flag or "O2"
+                return ApexMixedPrecisionPlugin(self._amp_level_flag)
 
         raise RuntimeError("No precision set")
 
-    def select_strategy(self) -> Strategy:
-        if isinstance(self.distributed_backend, Accelerator) and self.distributed_backend.strategy is not None:
-            plugin = self.distributed_backend.strategy
-        elif self.use_ddp2:
-            plugin = DDP2Strategy(parallel_devices=self.parallel_devices, cluster_environment=self.cluster_environment)
-        elif self.use_ddp and self.use_deepspeed:
-            plugin = DeepSpeedStrategy(
-                cluster_environment=self.select_cluster_environment(), parallel_devices=self.parallel_devices
+    def _precision_misconfig_check(self):
+        # TODO change exception type to ImpactableConfigurationException
+        if isinstance(self.accelerator, IPUAccelerator):
+            if self._precision_flag not in (16, 32):
+                raise MisconfigurationException(
+                    f"`Trainer(accelerator='ipu', precision={self._precision_flag!r})` is not supported."
+                )
```

Comment:
This error message could be moved inside `IPUPrecisionPlugin.__init__`.

In fact, each __init__ should contain its own input validation as long as it has all the required information to do it.

Reply:
I totally agree with you.  The compatibility/availability check should be moved to xClass __init__ as much as possible. For example, the device_avalibility_check should be in xAccelerator as well. 
But here, I think it's an RFC, as we need pass `_precision_flag` to PrecisionPlugin(). Do we prefer check here or pass into PrecisionPlugin and check in the PrecisionPlugin.__init__()?